### ISSUE TITLE
Override code generation not working

### ISSUE THREAD

**Issue by SubtleMetaphor** [OmniSharp/omnisharp-vscode#4600]
Issue Type: <b>Bug</b>

If there is a trailing white space ANYWHERE in a class, the override code generation will not work.

https://gyazo.com/be89e3cb0b6ffe9a86697ba1f5d0406e

Extension version: 1.23.12
VS Code version: Code 1.56.2 (054a9295330880ed74ceaedda236253b4f39a335, 2021-05-12T17:13:13.157Z)
OS version: Windows_NT x64 10.0.19042

**Comment by 333fred** [comment]
I'm unable to reproduce this issue:
![Demo](https://user-images.githubusercontent.com/2371880/125185821-cf7b8c00-e1db-11eb-9e54-dc6a2dd318ca.gif)

Can you give a more precise set of instructions? Do you have other extensions installed that could be interfering with the completion list?

**Comment by SubtleMetaphor** [comment]
> I'm unable to reproduce this issue:
> ![Demo](https://user-images.githubusercontent.com/2371880/125185821-cf7b8c00-e1db-11eb-9e54-dc6a2dd318ca.gif)
> 
> Can you give a more precise set of instructions? Do you have other extensions installed that could be interfering with the completion list?

Maybe it's a spaces vs tab issue? (I was using tabs.)

**Comment by 333fred** [comment]
> Maybe it's a spaces vs tab issue? (I was using tabs.)

I could try, but a precise set of instructions for reproducing the issue and your omnisharp logs are likely going to be the only way to make progress here.

**Comment by DArkHekRoMaNT** [comment]
Same problem. Override sometimes stops working. Moreover, it can work now, stop working in two minutes, and start working again in three minutes, some kind of madness. Previously, everything worked fine, but lately I see this more and more often.

**Comment by DArkHekRoMaNT** [comment]
Your conversation about spaces and tabs helped me find the reason. Override unexpectedly requires proper indentation o.O

I often do three enter and up arrow for a new method. As a result, the up arrow takes me not to the class level, unlike enter, but to the beginning of the line, I immediately write override (since I see no reason to add indentation, which will be fixed anyway) and it does not work.

**Comment by 333fred** [comment]
@DArkHekRoMaNT I'm still looking for a way to reproduce this. If you have info, please let me know.

**Comment by SirIntruder** [comment]
I've noticed this mysterious behaviour too - after some trial and error I realised it's not triggered by just any trailing whitespace anywhere, but for some reason it really hates trailing tab on class-scope opening brace. Tried this in omnisharp-roslyn repo

![2021-12-26_18-27-15](https://user-images.githubusercontent.com/5102661/147415638-414d46ad-b50e-4542-aebf-463d45e5a174.png)
![2021-12-26_18-27-02](https://user-images.githubusercontent.com/5102661/147415640-304bd9ab-cb5f-4c4d-b6f3-998724ec9ad8.png)

it goes together with these logs:

```
[warn]: OmniSharp.Roslyn.CSharp.Services.Navigation.FindUsagesService
        No symbol found. File: c:\Git\omnisharp\omnisharp-roslyn\src\OmniSharp.Cake\CakeContextModel.cs, Line: 4, Column: 15.
Received response for /v2/getcodeactions but could not find request.
```

**Comment by ddavisso4** [comment]
Seems to work for me as long as I start typing `public override` at the very end of the class or the very beginning. Anywhere in between gives the error below. It also seems to dislike improper indentation (e.g. if I start at the very beginning of a line it doesn't work even at the beginning or end of the class).

```
  "Request_seq": 390,
  "Command": "/completion",
  "Running": true,
  "Success": false,
  "Message": "\"System.ArgumentOutOfRangeException: Length cannot be less than zero.\\nParameter name: length\\n  at System.String.Substring (System.Int32 startIndex, System.Int32 length) [0x00031] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 \\n  at OmniSharp.Roslyn.CSharp.Services.Completion.CompletionListBuilder.<GetCompletionInfo>g__getPossiblySnippetizedInsertText|16_0 (Microsoft.CodeAnalysis.Text.TextChange change, System.Nullable`1[T] adjustedNewPosition) [0x00060] in <dbd4c42c78874a71919856b53db6870e>:0 \\n  at OmniSharp.Roslyn.CSharp.Services.Completion.CompletionListBuilder.GetCompletionInfo (Microsoft.CodeAnalysis.Text.SourceText sourceText, System.Int32 position, Microsoft.CodeAnalysis.Completion.CompletionItem completion, Microsoft.CodeAnalysis.Completion.CompletionChange change, Microsoft.CodeAnalysis.Text.TextSpan typedSpan, System.String labelText, System.Boolean expectingImportedItems, System.String& insertText, System.String& filterText, System.String& sortText, OmniSharp.Models.v1.Completion.InsertTextFormat& insertTextFormat, Microsoft.CodeAnalysis.Text.TextSpan& changeSpan, System.Collections.Generic.List`1[OmniSharp.Models.LinePositionSpanTextChange]& additionalTextEdits) [0x000d6] in <dbd4c42c78874a71919856b53db6870e>:0 \\n  at OmniSharp.Roslyn.CSharp.Services.Completion.CompletionListBuilder.BuildCompletionItemsSync (Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.SourceText sourceText, System.Int64 cacheId, System.Int32 position, Microsoft.CodeAnalysis.Completion.CompletionService completionService, Microsoft.CodeAnalysis.Completion.CompletionList completions, Microsoft.CodeAnalysis.Text.TextSpan typedSpan, System.Boolean expectingImportedItems, System.Boolean isSuggestionMode) [0x0021b] in <dbd4c42c78874a71919856b53db6870e>:0 \\n  at OmniSharp.Roslyn.CSharp.Services.Completion.CompletionListBuilder.BuildCompletionItems (Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.SourceText sourceText, System.Int64 cacheId, System.Int32 position, Microsoft.CodeAnalysis.Completion.CompletionService completionService, Microsoft.CodeAnalysis.Completion.CompletionList completions, Microsoft.CodeAnalysis.Text.TextSpan typedSpan, System.Boolean expectingImportedItems, System.Boolean isSuggestionMode, System.Boolean enableAsyncCompletion) [0x00138] in <dbd4c42c78874a71919856b53db6870e>:0 \\n  at OmniSharp.Roslyn.CSharp.Services.Completion.CompletionService.Handle (OmniSharp.Models.v1.Completion.CompletionRequest request) [0x005bb] in <dbd4c42c78874a71919856b53db6870e>:0 \\n  at OmniSharp.Endpoint.EndpointHandler`2[TRequest,TResponse].GetFirstNotEmptyResponseFromHandlers (OmniSharp.Endpoint.Exports.ExportHandler`2[TRequest,TResponse][] handlers, TRequest request) [0x00099] in <7c5a61c964114ca68a62c9da132f5af2>:0 \\n  at OmniSharp.Endpoint.EndpointHandler`2[TRequest,TResponse].HandleRequestForLanguage (System.String language, TRequest request, OmniSharp.Protocol.RequestPacket packet) [0x00163] in <7c5a61c964114ca68a62c9da132f5af2>:0 \\n  at OmniSharp.Endpoint.EndpointHandler`2[TRequest,TResponse].Process (OmniSharp.Protocol.RequestPacket packet, OmniSharp.Endpoint.LanguageModel model, Newtonsoft.Json.Linq.JToken requestObject) [0x0024b] in <7c5a61c964114ca68a62c9da132f5af2>:0 \\n  at OmniSharp.Stdio.Host.HandleRequest (System.String json, Microsoft.Extensions.Logging.ILogger logger) [0x000f3] in <a50105da3f9c45b299f1e289aba046fa>:0 \"",
  "Body": null,
  "Seq": 1047,
  "Type": "response"
}
```

**Comment by 333fred** [comment]
Thanks for the help @SirIntruder, I was able to reproduce from your pictures, and have a bug fix in #2319. @ddavisso4, I _don't_ think this addresses the issue you mentioned: I didn't see anything about an ArgumentOutOfRange exception in the repro for this bug. If you encounter that again, please file a separate issue with a set of instructions on how to reproduce it.
